### PR TITLE
Patternfly5

### PR DESCRIPTION
**Pull request by willr3**
## Fixes Issue
Migrates to patternfly5 using the `deprecated` patternfly package where needed. We will still need to migrate off the deprecated package.

Fixes: #714

### CODE REVIEW

File: horreum-web/package.json [hunk 1]
```diff
@@ -4,9 +4,9 @@
   "private": true,
   "dependencies": {
     "@monaco-editor/react": "4.6.0",
-    "@patternfly/patternfly": "4.224.4",
-    "@patternfly/react-core": "4.277.0",
-    "@patternfly/react-table": "4.113.5",
+    "@patternfly/patternfly": "^5.1.0",
+    "@patternfly/react-core": "^5.1.1",
+    "@patternfly/react-table": "^5.1.1",
```
Comment (shivam-sharma7) [inline horreum-web/package.json:9]:
```suggestion
    "@patternfly/patternfly": "5.1.0",
    "@patternfly/react-core": "5.1.1",
    "@patternfly/react-table": "5.1.1",
```
please, use specific versions

**Review by shivam-sharma7 — CHANGES_REQUESTED** [review]

File: horreum-web/package.json (hunk 1 above)
Comment (willr3) [reply horreum-web/package.json:9]: This is for a UI library, is there an issue with putting the version range in package.json and relying on package-lock to preserve the specific version?

Comment (willr3) [reply horreum-web/package.json]: I made the change and see `"vite-tsconfig-paths": "^4.2.3"` in `master` is that also being updated to follow a specific version?
https://github.com/Hyperfoil/Horreum/blob/master/horreum-web/package.json#L49

Comment (johnaohara) [reply horreum-web/package.json]: it should be yes

Comment (johnaohara) [reply horreum-web/package.json]: > This is for a UI library, is there an issue with putting the version range in package.json and relying on package-lock to preserve the specific version?

Until we have unit tests for the UI, this limits the probability of in compatible transient deps breaking things. It isn't ideal, but neither is having a lack of UI testing. We can revisit this once we have UI tests that will detect breaking changes

Comment (shivam-sharma7) [reply horreum-web/package.json]: @willr3 make sure your lock file is up to date with your current change

**Comment by johnaohara** [comment]
I was going to backport this, but given the risk of introducing problems, I am going to hold off until the 0.12 release to give more time to test

**Comment by johnaohara** [comment]
@willr3 this looks good now, thanks for fixing the issues.

Please can you squash and rebase, thanks

**Comment by johnaohara** [comment]
Thanks a lot for your contribution!

**Pull request merged**